### ISSUE TITLE
CantPackException: bad DT_STRSZ 0x245c [C#]

### ISSUE THREAD

**Issue by misiek08** [upx/upx#858]
## What's the problem (or question)?
I'm getting error instead of compressed binary.
```
                       Ultimate Packer for eXecutables
                          Copyright (C) 1996 - 2024
UPX 4.2.4       Markus Oberhumer, Laszlo Molnar & John Reiser    May 9th 2024

        File size         Ratio      Format      Name
   --------------------   ------   -----------   -----------
upx: StaticFileServer: CantPackException: bad DT_STRSZ 0x245c

Packed 0 files.
```

## What should have happened?
I should get compressed file, with `Packed 1 file.` info, example:
```
                       Ultimate Packer for eXecutables
                          Copyright (C) 1996 - 2024
UPX 4.2.4       Markus Oberhumer, Laszlo Molnar & John Reiser    May 9th 2024

        File size         Ratio      Format      Name
   --------------------   ------   -----------   -----------
   6778395 ->   3859192   56.93%   linux/amd64   controller

Packed 1 file.
```

## Do you have an idea for a solution?
Didn't investigate. It worked great for Golang binary, but didn't work for C#/.NET8.0 PublishSingleFile=true.

## How can we reproduce the issue?
1. Prepare simplest .NET 8.0 C# sample program. I have simple static file server.
2. `dotnet publish -p:PublishSingleFile=true`
3. `upx -9 <binary>`
4. Get error

## Please tell us details about your environment.

* UPX version used (`upx --version`): 4.2.4
* Host Operating System and version: Ubuntu 22.04
* Host CPU architecture: amd64
* Target Operating System and version: Ubuntu 22.04
* Target CPU architecture: amd64

**Comment by jreiser** [comment]
Please upload ("Paste, drop, or click to add files"), or give a downloadable URL, of an actual executable binary file that UPX declines to compress.  Supplying an actual file is so much simpler and more reliable than any recipe to reproduce.  GitHub accepts `.zip`, `.gz`, or ` .tgz` files, so apply `zip`, `gzip`, or `tar -cz` before uploading.

**Comment by misiek08** [comment]
[server.tar.gz.partaa.gz](https://github.com/user-attachments/files/17896039/server.tar.gz.partaa.gz)
[server.tar.gz.partab.gz](https://github.com/user-attachments/files/17896042/server.tar.gz.partab.gz)

Those are not directly .gz files. I did split the original tar.gz archive and just added the extension so GH will be happy.
```
tar czvf server.tar.gz StaticFileServer
split -b 25M server.tar.gz  "server.tar.gz.part"
mv server.tar.gz.partaa server.tar.gz.partaa.gz
mv server.tar.gz.partab server.tar.gz.partab.gz
```

**Comment by jreiser** [comment]
I re-constructed `StaticFileServer` by
```
$  cat server.tar.gz.partaa.gz server.tar.gz.partab.gz  |  gzip -d  |  tar xvf -
StaticFileServer
$ ls -l StaticFileServer
-rwxr-xr-x. 1 user group 92368004 Nov  5 15:05 StaticFileServer
$ sha256sum StaticFileServer
c3bd7b851025d8a3633c33f97f76a2792d64247963c5f1ada4d4ed6f15f84e04  StaticFileServer
```

This bug in PackLinuxElf64::invert_pt_dynamic() was fixed by commit 937ae4c5e on 2024-06-04.
You can download a build from https://github.com/upx/upx/actions/runs/11239291138 , such as `upx-jreiser-pagebig-571c9d1-ubuntu-24.04.zip` (or perhaps -ubuntu-22.04 if you do not have the latest Ubuntu.)
After unzipping, look at `zstdcat upx-jreiser-pagebig-571c9d1-ubuntu-22.04.tar.zst | tar tvf - | grep 'upx$'`, then extract your favorite executable with something such as `zstdcat upx-jreiser-pagebig-571c9d1-ubuntu-22.04.tar.zst | tar xvf - upx-jreiser-pagebig-571c9d1-ubuntu-22.04/InstallWithCMake/InstallWithDestdir/gcc/debug/usr/local/bin/upx`.

**Comment by misiek08** [comment]
`c3bd7b851025d8a3633c33f97f76a2792d64247963c5f1ada4d4ed6f15f84e04  StaticFileServer`
correct :)

When there will be new version released? I just used 4.2.4 that is the newest :)
I'll try to use the binary from actions, but it would be great to see release ;)
